### PR TITLE
networking fixes and configuration file cleanup

### PR DESCRIPTION
**Pull request by justinjpacheco**
- apt
    - added apt attributes file for apt specific configuration
    - updated apt.conf.d/00-bcpc to accept all apt config as template
      variable
    - added support for customized sources.list contents
  - calico
    - removed old calico config
    - moved calico config from default into attributes/calico.rb
    - added arch amd64 to apt_repository resource
  - networking
    - fixed incorrect array indexing from regex capture
    - moved systemd-resolved dns configuration to networking.rb recipe
  - mysql
    - added arch amd64 to apt_repository resource
  - systemd-resolved
    - added configuration option for fallback dns servers
  - virtual environment
    - removed proxy,dns_servers and ntp configuration. defaults are now in
      defaults.rb

### CODE REVIEW

File: chef/cookbooks/bcpc/attributes/default.rb [hunk 1]
```diff
@@ -9,7 +9,12 @@
 default['bcpc']['cloud']['region'] = node.chef_environment
 default['bcpc']['cloud']['vip'] = {'ip': '10.10.254.254/32'}
 
+# list of dns servers to use
 default['bcpc']['dns_servers'] = ["8.8.8.8", "8.8.4.4"]
+
+# list of ntp servers to use
+default['bcpc']['ntp']['servers'] = ["time.google.com"]
```
Comment (bryanapperson) [inline chef/cookbooks/bcpc/attributes/default.rb]: Only one timeserver?

Comment (justinjpacheco) [reply chef/cookbooks/bcpc/attributes/default.rb]: one by default, it can take an array if the user wants to add more

Comment (mihalis68) [reply chef/cookbooks/bcpc/attributes/default.rb]: a real datacenter cluster would do better

Comment (justinjpacheco) [reply chef/cookbooks/bcpc/attributes/default.rb]: @mihalis68 we can add time1.google.com, time2.google.com as defaults if you like

Comment (mihalis68) [reply chef/cookbooks/bcpc/attributes/default.rb]: sure, hinting to people to use more than 1 is good practice

File: chef/cookbooks/bcpc/recipes/calico-apt.rb [hunk 2]
```diff
@@ -37,6 +37,7 @@
 return unless node['bcpc']['calico']['repo']['enabled']
 
 apt_repository "calico" do
+  arch 'amd64'
   uri node['bcpc']['calico']['repo']['url']
   distribution 'xenial'
```
Comment (comay) [inline chef/cookbooks/bcpc/recipes/calico-apt.rb]: Is there a 'bionic' version available for Calico?

File: chef/cookbooks/bcpc/attributes/default.rb [hunk 3]
```diff
@@ -19,6 +24,16 @@
...
 vip = IPAddress(node['bcpc']['cloud']['vip']['ip']).address
 
+###############################################################################
+# ubuntu
+###############################################################################
+
+default['bcpc']['ubuntu']['archive_url'] = 'http://archive.ubuntu.com/ubuntu'
+default['bcpc']['ubuntu']['security_url'] = 'http://security.ubuntu.com/ubuntu'
+default['bcpc']['ubuntu']['codename'] = node['lsb']['codename']
+default['bcpc']['ubuntu']['components'] = ['main','restricted','universe','multiverse']
```
Comment (comay) [inline chef/cookbooks/bcpc/attributes/default.rb]: Spaces after each comma?

File: chef/cookbooks/bcpc/recipes/calico-apt.rb (hunk 2 above)
Comment (justinjpacheco) [reply chef/cookbooks/bcpc/recipes/calico-apt.rb]: there does not appear to be a bionic version for calico yet

File: chef/cookbooks/bcpc/attributes/default.rb (hunk 3 above)
Comment (justinjpacheco) [reply chef/cookbooks/bcpc/attributes/default.rb]: i will address these and others when i implement pre-commit hooks for linting ruby files

**Comment by justinjpacheco** [comment]
Documentation seems like the best choice at the moment. Something like a "howto" section showing examples of common things to override.

**Pull request merged**